post_deployment_tests/deployment_test.py

```diff
@@ -0,0 +1,338 @@
+        self.assertEqual(data['message'], 'File added to archive')
+
+    def test_simple_status(self):
+        """Test statusing a simple text file."""
+        fileid = '1234'
+        resp = requests.head(str(ARCHIVEURL + fileid))
+        self.assertEqual(resp.status_code, 204)
+        self.assertEqual(resp.headers['x-pacifica-file-storage-media'], 'disk')
+        self.assertEqual(resp.headers['content-length'], '30')
+        self.assertEqual(resp.headers['x-pacifica-messsage'], 'File was found')
+
+    def test_simple_stage(self):
+        """test staging a simple text file."""
+        fileid = '1234'
+        resp = requests.post(str(ARCHIVEURL + fileid))
+        self.assertEqual(resp.status_code, 200)
+        data = resp.json()
+        self.assertEqual(data['message'], 'File was staged')
+
+    def test_simple_read(self):
+        """test reading a simple text file."""
+        fileid = '1234'
+        filename = '/tmp/test_simple_read.txt'
+        resp = requests.get(str(ARCHIVEURL + fileid), stream=True)
+        myfile = open(filename, 'wb+')
+        buf = resp.raw.read(1024)
+        while buf:
+            myfile.write(buf)
+            buf = resp.raw.read(1024)
+        myfile.close()
+        self.local_files[filename] = filename
+        filesize = os.path.getsize(filename)
+        # know the simple file writtten is 30 bytes from archive
+        self.assertEqual(filesize, 30)
+
+    def test_file_rewrite(self):
+        """Test trying to rewrite a file, rewrite should fail."""
+        filename = '/tmp/test_simple_write.txt'
+        fileid = '1234'
+        file1 = open(filename, 'w+')
+        file1.write('Writing content for first file')
+        file1.close()
+        f = open(filename, 'rb')
+        resp = requests.put(str(ARCHIVEURL + fileid), data=f)
+        f.close()
+        self.assertEqual(resp.status_code, 500)
+        data = resp.json()
+        error_msg = 'Can\'t open'
+        # get error message length since the file path returned is unique per deploymnet while
+        # the rest of the error message is not
+        err_msg_length = len(error_msg)
+        self.assertEqual(data['message'][:err_msg_length], error_msg)
+
+    def test_simple_cleanup(self):
+        """Clean up files that are created for testing."""
+        if CLEANLOCALFILES:
+            for filepath in self.local_files:
+                os.remove(filepath)
+
+        if CLEANARCHIVEFILES:
+            for filepath in self.archive_files:
+                os.remove(ARCHIVEPREFIX + filepath)
+        self.assertEqual(True, True)
+
+
+class BinaryFileArchiveTests(unittest.TestCase):
+    """Class for testing binary files through the archive workflow."""
+
+    local_files = {}
+    archive_files = {}
+
+    def test_binary_file_write(self):
+        """Write a binary file to the archive."""
+        filename = '/tmp/binary_file'
+        fileid = '4321'
+        newFileBytes = [123, 3, 255, 0, 100]
```

Comment:
Either check the content of the binary file or just use something random `os.random()`